NC-NEWS/src/components/TopicsArticles.jsx

```diff
@@ -1,33 +1,24 @@
 import { useEffect, useState } from "react";
-import { Link, useParams } from "react-router-dom";
-import { getArticlesByTopic } from "../api";
+import { Link, useParams } from "react-router-dom"; 
+import { getArticlesByTopic } from "../api"; 
 import ArticlesCard from "./ArticlesCard";
 
 const TopicsArticles = ({ topic, topicsURL }) => {
   const { slug } = topic;
   const [articlesData, setArticlesData] = useState([]);
   const [isLoading, setIsLoading] = useState(true);
 
-  useEffect(() => {
-    const fetchArticlesByTopic = () => {
-      setIsLoading(true);
 
-      return new Promise((resolve, reject) => {
-        getArticlesByTopic(slug)
-          .then((articles) => {
-            setArticlesData(articles);
-            setIsLoading(false);
-            resolve(articles);
-          })
-          .catch((error) => {
-            setIsLoading(false);
-            reject(error);
-          });
-      });
-    };
 
-    fetchArticlesByTopic();
-  }, [slug]);
+  useEffect(() => {
+    getArticlesByTopic(slug)
+      .then((articles) => {
+        setArticlesData(articles);
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  }, [articlesData]);
```

Comment:
I would expect slug to be in the dependancy array not articlesData, when the slug changes is when we would want to refetch the articles